onnxruntime/test/providers/cpu/nn/conv_fp16_test.cc

```diff
@@ -37,6 +37,12 @@ void TestConvFp16Op(const ConvOpAndTestAttributes& attributes,
                     OpTester::ExpectResult expect_result = OpTester::ExpectResult::kExpectSuccess,
                     const std::string& err_str = "",
                     int opset = 11) {
+#if !defined(MLAS_F16VEC_INTRINSICS_SUPPORTED)
+  // a `return` after tester will make binary crash
```

Comment:
Is the reason that if you do not have MLAS_F16VEC_INTRINSICS_SUPPORTED there's no support for the custom NhwcFusedConv attribute? And now that we hit this code when the CoreML EP is enabled we need to do an early exit in that case as no EP can handle the node?

Reply:
Yeah.  If there is no EP can handle the node, a instance of OpTester will crash. And here we have no way to get avaible EPs.

Reply:
My question was more about making the comment explain 'why' rather than 'what' will happen. If you say 'it will crash' the next developer to look at the code has to start from scratch to figure out the reason in order to decide if this applies to their potential changes not.

I think it comes down to the activation attribute being something that the FusedConv/NhwcFusedConv operators have that the ONNX Conv doesn't. As the CoreML EP doesn't support those contrib ops, if the CPU EP can't handle it due to the lack of MLAS support the OpTester won't find any EPs that can run and fails.

i.e. if attributes.activation is not empty we're using a contrib op that only a small number of EPs support.

I'm surprised this check isn't also required for the QNN EP as I don't see FusedConv or NhwcFusedConv in the ops it supports.

Reply:
Got it.

In the current main branch,  This test will be compiled and ran iff `MLAS_F16VEC_INTRINSICS_SUPPORTED`  is defined. For those has QNN EP, I think it is not touch at all.

Basiclly, we thought that CPU EP can always take in any Op which is not supportted by other EPs, however, this is not the case. Then we have lots of MACROs to control which test should run on this platform.

I think CoreML EP is such a case where CoreML support pars of Ops (attributes is empty) and refused others. 

But yes, the reason is as you talked that no EPs can accept NhwcFusedConv in fp16 with activation fused in.

Reply:
Comments added